### ISSUE TITLE
Auto fmt on save?

### ISSUE THREAD

**Issue by soulshake** [z0mbix/vim-shfmt#1]
Thanks, this is a useful plugin!

It would be super awesome if this could (optionally) auto format when a file is saved, similar to `vim-terraform`'s [`g:terraform_fmt_on_save`](https://github.com/hashivim/vim-terraform#overview-vim-terraform).

**Comment by z0mbix** [comment]
I did get this working but never committed/documented it. I'll take another look in to it when I can find some free time.

**Comment by z0mbix** [comment]
@soulshake sorry it took a while, can you please let me know if it works OK for you?

`let g:shfmt_fmt_on_save = 1`

![Works on my machine](https://rlv.zcache.com/works_on_my_machine_classic_round_sticker-r56ce1cc314be46efbe749e9c58c761d5_v9waf_8byvr_540.jpg)

I've also standardised the variable naming so if you were using`vimshfmt_extra_args`, please update to just `shfmt_extra_args`.

**Comment by soulshake** [comment]
It works! :tada: 

I was using `let g:shfmt_switches = ['-i 2']`. Once I replaced this with `let g:shfmt_extra_args = '-i 2'` it worked as expected.

I'll use it for a while and let you know if I notice any unexpected behavior.

Thank you!

**Comment by z0mbix** [comment]
Excellent!

**Comment by z0mbix** [comment]
I’ll fix the README tomorrow, can’t belive I missed that!

**Comment by alecandido** [comment]
Sorry to necrobump: for some reason here it's not working.

Maybe I'm stupid and I made some trivial mistake, but I simply put in my `.vim/ftplugin/sh.vim`:
```vim
let g:shfmt_fmt_on_save = 1

augroup LocalShell
    autocmd!

    autocmd BufWritePre * Shfmt
augroup END
```

The first one alone it's not working, the second one yes (of course even alone).

Tested with both `vim` and `nvim`, the plugin has been installed with `vim-plug`. If I'll try to fix I'll update as soon as I find something useful.
